packages/keycloak-user-management/src/components/UserGroupDetailView/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { Col, Space, Typography, Spin } from 'antd';
+import { CloseOutlined } from '@ant-design/icons';
+import { useHistory } from 'react-router';
+import { Resource404 } from '@opensrp/react-utils';
+import { sendErrorNotification } from '@opensrp/notifications';
+import { Button } from 'antd';
+import { URL_USER_EDIT, URL_USER_GROUPS } from '../../constants';
+import { UserGroupMembers } from '../UserGroupsList';
+import { KeycloakUserGroup } from 'keycloak-user-management/src/ducks/userGroups';
+import { Link } from 'react-router-dom';
+import { loadGroupDetails, loadGroupMembers } from '../UserGroupsList/utils';
+import { ERROR_OCCURED, MEMBERS, NAME, ROLES } from '../../lang';
+const { Text } = Typography;
+
+/** typings for the view details component */
+export interface ViewDetailsProps {
+  groupId: string;
+  keycloakBaseURL: string;
+}
+export const defaultProps = {
+  groupId: '',
+};
+/** component that renders the details view to the right side
+ * of list view
+ *
+ * @param props - detail view component props
+ */
+const ViewDetails = (props: ViewDetailsProps) => {
+  const { groupId, keycloakBaseURL } = props;
+  const history = useHistory();
+  const [loading, setLoading] = React.useState<boolean>(true);
+  const [userGroupMembers, setUserGroupMembers] = React.useState<UserGroupMembers[] | null>(null);
+  const [singleUserGroup, setSingleUserGroup] = React.useState<KeycloakUserGroup | null>(null);
+
+  React.useEffect(() => {
+    if (groupId) {
+      setLoading(true);
+      const membersPromise = loadGroupMembers(groupId, keycloakBaseURL, setUserGroupMembers);
+      const userGroupPromise = loadGroupDetails(groupId, keycloakBaseURL, setSingleUserGroup);
+      Promise.all([membersPromise, userGroupPromise])
+        .catch(() => sendErrorNotification(ERROR_OCCURED))
+        .finally(() => setLoading(false));
```

Comment:
i found this way of setting the loading to be not that much effective as for me i prefer 

1) put a actual if condition instead of if (loading) e.g for data we should do like if (!data1 || !data 2 || !data3)

2) use Promise.all to set loading value to be true 

i personally recommend the first one as in this way we are 100% sure that only then the sub  component is loaded when the data is ready in all format and the sub component is loaded only once and is responsible for the data manipulation not data fetch or load

Reply:
@ashfahan for option 1, if i click on the detail view for group x, then set data fetched for that group to state, if i add the above condition you suggested for checking if data is null/undefined how will i load data for the next group (say group y) because group x data will still be there.

Reply:
>  be not that much effective

how was it not effective, could you expound on this.

in 1. does it mean we woulld be showing a loader even if some data was fetch-resolved to a falsy value, do you think this semantically represents a correct ux, given the fact that in such a case the loader would persist indefinitely.



> 100% sure that only then the sub component is loaded when the data is ready in all format

IMO, if you have some data that is falsy and is **needed for components render**, we can catch that early and show the user an error page. e.g
```ts
if (isLoading) return <Loader/>
if (!data) return <NotFound />
```
the focus being on *needed*. if the component's render can still work with the empty or falsy value of the data, then we do not need the early null/missing-data check

Reply:
@ashfahan I also support we check `isLoading` status to display the loader since `null` value can be as a result of many other things

Reply:
its the same situation like we had in location module in that in order to keep the logic and data loading separate from the view, i was passing the event from the table back to HOC component that was responsible for loading the component and then as the data loads it mount the details component


https://github.com/OpenSRP/web/blob/bd0ee1f37704bc9f3c3951949ad5878199c68930/packages/location-management/src/components/LocationUnitView/index.tsx#L58-L78